src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -393,7 +452,317 @@ public virtual bool TryInsertExpansion(int startPositionInSubjectBuffer, int end
+            }
+
+            var document = SubjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+            if (document is null)
+            {
+                return;
+            }
+
+            var newSymbolKey = (e.SelectedItem as AbstractSignatureHelpProvider.SymbolKeySignatureHelpItem)?.SymbolKey ?? default;
+            var newSymbol = newSymbolKey.Resolve(document.Project.GetRequiredCompilationAsync(CancellationToken.None).WaitAndGetResult(CancellationToken.None), cancellationToken: CancellationToken.None).GetAnySymbol();
+            if (newSymbol is not IMethodSymbol method)
+                return;
+
+            MoveToSymbol(_state._symbols, method, CancellationToken.None);
+        }
+
+        private async Task<ImmutableArray<ISymbol>> GetSymbolsAsync(
+            SnapshotPoint caretPosition,
+            CancellationToken cancellationToken)
+        {
+            var document = SubjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+            if (document is null)
+            {
+                return ImmutableArray<ISymbol>.Empty;
+            }
+
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var token = await semanticModel.SyntaxTree.GetTouchingTokenAsync(caretPosition.Position, cancellationToken).ConfigureAwait(false);
+            var semanticInfo = semanticModel.GetSemanticInfo(token, document.Project.Solution.Workspace, cancellationToken);
+            return semanticInfo.ReferencedSymbols;
+        }
+
+        public void MoveToSymbol(ImmutableArray<IMethodSymbol> symbols, IMethodSymbol symbol, CancellationToken cancellationToken)
+        {
+            AssertIsForeground();
+
+            if (ExpansionSession is null)
+            {
+                return;
+            }
+
+            if (SymbolEqualityComparer.Default.Equals(_state._symbol, symbol))
+            {
+                return;
+            }
+
+            var symbolName = _state._symbol?.Name ?? _state._symbols.FirstOrDefault()?.Name;
+            if (symbolName != symbol.Name)
+            {
+                // Can't track this request
+                _state.ClearSymbolInformation();
+                return;
+            }
+
+            if (!_state._symbols.IsDefaultOrEmpty && _state._symbols != symbols)
+            {
+                // Can't track this request
+                _state.ClearSymbolInformation();
+                return;
+            }
+
+            var textViewModel = TextView.TextViewModel;
+            if (textViewModel == null)
+            {
+                Debug.Assert(TextView.IsClosed);
+                return;
+            }
+
+            var buffer = EditorAdaptersFactoryService.GetBufferAdapter(textViewModel.DataBuffer);
+            if (buffer is not IVsExpansion expansion)
+            {
+                return;
+            }
+
+            // Track current argument values
```

Comment:
an example would be helpful here.  

i think i just got what's going on.  I think you're saying: if the user goes between overloads.  And one overload and another have "teh same parameter" (defined as same type and same name), then grab the value hte user has currently inserted in the text, and copy it into the new snippet.  If that's what's going on, please doc as such for clarity.

--

Aside: open questions about this.  If you are cycling between overloads, and you go from an overload with M parameters to N (where N is less than M), do you somehow still keep track of the parameters in the original overload not present in the new one?  That way you can reinsert them if the user cycles back around again?

Reply:
> do you somehow still keep track of the parameters in the original overload not present in the new one

Yes, and there are integration tests covering this case.